src/common/numformatter.cpp

```diff
@@ -171,6 +171,47 @@ bool wxNumberFormatter::GetThousandsSeparatorIfUsed(wxChar *sep)
 #endif // wxUSE_INTL/!wxUSE_INTL
 }
 
+bool wxNumberFormatter::GetThousandsSeparatorAndGroupingIfUsed(wxChar *sep, wxString *gr)
+{
+#if wxUSE_INTL
+    static wxChar s_thousandsSeparator = 0;
+    static wxString s_grouping;
+    static LocaleId s_localeUsedForInit;
+
+    if ( s_localeUsedForInit.NotInitializedOrHasChanged() )
+    {
+        const wxString
+            s = wxLocale::GetInfo(wxLOCALE_THOUSANDS_SEP, wxLOCALE_CAT_NUMBER);
+        if ( s.length() == 1 )
+        {
+            s_thousandsSeparator = s[0];
+            const wxString
+                g = wxLocale::GetInfo(wxLOCALE_GROUPING, wxLOCALE_CAT_NUMBER);
+            if ( g[0] != '\0')
```

Comment:
I don't understand this test... First of all, I think we should always overwrite `s_grouping`, as it can have the value corresponding to the old locale. Second, AFAICS `g` will never have a NUL byte inside it.

Reply:
Yes, I think it is unnecessary. I will remove it.

Reply:
Fixed in 9ed8ad9a90